recipes/cryptopp/all/conanfile.py

```diff
@@ -179,6 +189,12 @@ def package_info(self):
         elif self.settings.os == "Windows":
             self.cpp_info.components["libcryptopp"].system_libs = ["bcrypt", "ws2_32"]
 
+        if not self.options.shared and self.options.use_openmp:
+            if self.settings.compiler in ("gcc", "clang"):
+                openmp_flag = ["-fopenmp"]
+                self.cpp_info.components["libcryptopp"].cxxflags = openmp_flag
+                self.cpp_info.components["libcryptopp"].exelinkflags = openmp_flag
```

Comment:
@jcar87 Why not include MSVC and apple-clang support as well? Why `exelinkflags` but not `sharedlinkflags`?

`cxxflags` should be left unset as none of the public headers use `#pragma omp` or `#include <omp.h>` in their sources (it's not marked as such in the list of OpenMP consumers here at least: https://github.com/conan-io/conan-center-index/issues/24577#issuecomment-2241199222). And if it were necessary, then the `if not self.options.shared` part should be omitted.

Reply:
indeed it is `sharedlinkflags`, thanks.


MSVC is not needed here because `/openmp` is a compiler flag, not a linker one:
```
LINK : warning LNK4044: unrecognized option '/openmp'; ignored 
```
MSVC knows to link the runtime when one of the objects being linked require it (because they were compiled with `/openmp`) - fairly certain msvc embeds some directive in the obj that then gets processed by the linker - so no need to handle anything for msvc in this case.

OpenMP support in apple-clang is not documented by Apple, likely not officially supported at all - and the runtime is externally provided. We're just ensuring this works in the most likely used scenario - while the issue at large remains to be addressed.